sdk/core/src/metadata.rs

```diff
@@ -1,34 +1,43 @@
 use crate::AddAsHeader;
+use bytes::Bytes;
 use http::request::Builder;
 use std::collections::HashMap;
 
 #[derive(Debug, Clone)]
-pub struct Metadata(HashMap<String, String>);
+pub struct Metadata(HashMap<String, Bytes>);
 
 impl Metadata {
     pub fn new() -> Self {
         Self(HashMap::new())
     }
 
-    pub fn as_mut(&mut self) -> &mut HashMap<String, String> {
+    pub fn as_mut(&mut self) -> &mut HashMap<String, Bytes> {
         &mut self.0
     }
 
-    pub fn insert<K, V>(&mut self, k: K, v: V) -> Option<String>
+    pub fn insert<K, V>(&mut self, k: K, v: V) -> Option<Bytes>
     where
         K: Into<String>,
-        V: Into<String>,
+        V: Into<Bytes>,
     {
         self.0.insert(k.into(), v.into())
     }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn get(&self, k: &str) -> Option<Bytes> {
+        self.0.get(k).map(|b| b.clone())
```

Comment:
Is there a reason to eagerly clone instead of returning a `&Bytes` and letting the user decide? I guess cloning `Bytes` is fairly inexpensive....